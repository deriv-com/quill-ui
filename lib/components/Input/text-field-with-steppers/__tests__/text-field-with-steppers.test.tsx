import { render, screen } from "@testing-library/react";
import TextFieldWithSteppers, { TextFieldWithSteppersProps } from "..";
import userEvent from "@testing-library/user-event";
import React from "react";

describe("TextFieldWithSteppers", () => {
    const setup = (props: Partial<TextFieldWithSteppersProps> = {}) => render(<TextFieldWithSteppers {...props} />);

    it("should match snapshot", () => {
        const { container } = setup({ placeholder: "Enter text" });
        expect(container).toMatchSnapshot();
    });

    it("should render with a label", () => {
        setup({ label: "Username" });
        expect(screen.getByLabelText("Username")).toBeInTheDocument();
    });

    it("should render disabled input", () => {
        setup({ label: "Username", disabled: true });
        expect(screen.getByLabelText("Username")).toBeDisabled();
    });

    it("should handle the hover for outline variant", async () => {
        const onHover = jest.fn();
        setup({
            placeholder: "Placeholder",
            onMouseEnter: onHover,
            variant: "outline",
        });
        const input = screen.getByPlaceholderText("Placeholder");
        await userEvent.hover(input);
        expect(onHover).toHaveBeenCalledTimes(1);
        expect(input.parentElement).toMatchSnapshot();
    });

    it("should display message when provided", () => {
        const { container } = setup({
            message: "Error message",
            status: "error",
        });
        expect(screen.getByText("Error message")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should display unitLeft when provided", () => {
        const { container } = setup({
            unitLeft: "$",
        });
        expect(screen.getByText("$")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should display unitRight when provided", () => {
        const { container } = setup({
            unitRight: "USD",
        });
        expect(screen.getByText("USD")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
