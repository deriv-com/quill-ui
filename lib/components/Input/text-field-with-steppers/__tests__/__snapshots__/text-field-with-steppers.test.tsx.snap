// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldWithSteppers should display message when provided 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--error quill-input__wrapper__size--lg"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-5"
            placeholder="0.00"
            type="text"
            value=""
          />
        </div>
        <span
          class="icon_wrapper status-icon--error"
        >
          <svg
            height="24"
            role="img"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.688 9.82 7.953 22.516c-.078.117-.078.195-.078.312 0 .313.234.547.547.547h15.117c.313 0 .586-.234.586-.547 0-.117-.04-.195-.117-.312L16.273 9.82A.29.29 0 0 0 16 9.625c-.156 0-.234.078-.312.195m-1.602-.976A2.25 2.25 0 0 1 16 7.75c.742 0 1.484.43 1.875 1.094l7.734 12.695c.235.39.391.82.391 1.29 0 1.327-1.094 2.421-2.46 2.421H8.421A2.434 2.434 0 0 1 6 22.828q0-.703.352-1.289zm3.164 12.031c0 .703-.586 1.25-1.25 1.25-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25m-.312-7.187v3.75c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937v-3.75c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938"
            />
          </svg>
        </span>
        <span
          class="icon_wrapper"
        >
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M12.188 10.75a.45.45 0 0 1-.438.438H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h10.5a.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="6960c367eebbbbd16d788a7ec242e181__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M6.938 5.5v4.813h4.812a.47.47 0 0 1 .438.437.45.45 0 0 1-.438.438H6.938V16a.45.45 0 0 1-.438.438.43.43 0 0 1-.437-.438v-4.812H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h4.813V5.5a.45.45 0 0 1 .437-.437.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="49593f687d6be52155263aa4ff7f7bf0__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </span>
      </div>
    </div>
    <div
      class="message__container message__container--lg message__container__status--error"
    >
      <span
        class="message__container__text"
      >
        Error message
      </span>
    </div>
  </div>
</div>
`;

exports[`TextFieldWithSteppers should display unitLeft when provided 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--lg"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <span
            class="quill-typography quill-typography__body-text__size--lg__weight--regular__decoration--default quill-typography__color--disabled quill-input-label__label quill-input-label__label--left"
          >
            $
          </span>
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-6"
            placeholder="0.00"
            type="text"
            value=""
          />
        </div>
        <span
          class="icon_wrapper"
        >
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M12.188 10.75a.45.45 0 0 1-.438.438H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h10.5a.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="6960c367eebbbbd16d788a7ec242e181__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M6.938 5.5v4.813h4.812a.47.47 0 0 1 .438.437.45.45 0 0 1-.438.438H6.938V16a.45.45 0 0 1-.438.438.43.43 0 0 1-.437-.438v-4.812H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h4.813V5.5a.45.45 0 0 1 .437-.437.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="49593f687d6be52155263aa4ff7f7bf0__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextFieldWithSteppers should display unitRight when provided 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--lg"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-7"
            placeholder="0.00"
            type="text"
            value=""
          />
          <span
            class="quill-typography quill-typography__body-text__size--lg__weight--regular__decoration--default quill-typography__color--default quill-input-label__label quill-input-label__label--right"
          >
            USD
          </span>
        </div>
        <span
          class="icon_wrapper"
        >
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M12.188 10.75a.45.45 0 0 1-.438.438H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h10.5a.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="6960c367eebbbbd16d788a7ec242e181__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M6.938 5.5v4.813h4.812a.47.47 0 0 1 .438.437.45.45 0 0 1-.438.438H6.938V16a.45.45 0 0 1-.438.438.43.43 0 0 1-.437-.438v-4.812H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h4.813V5.5a.45.45 0 0 1 .437-.437.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="49593f687d6be52155263aa4ff7f7bf0__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextFieldWithSteppers should handle the hover for outline variant 1`] = `
<div
  class="quill-input-no-label__wrapper"
>
  <input
    class="input peer input__align--left input__size--lg"
    id="quill-input-4"
    placeholder="Placeholder"
    type="text"
    value=""
  />
</div>
`;

exports[`TextFieldWithSteppers should match snapshot 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--lg"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--lg"
            id="quill-input-1"
            placeholder="Enter text"
            type="text"
            value=""
          />
        </div>
        <span
          class="icon_wrapper"
        >
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M12.188 10.75a.45.45 0 0 1-.438.438H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h10.5a.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="6960c367eebbbbd16d788a7ec242e181__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            class="quill-button quill-icon-button__size--md quill__color--tertiary-black-white"
            data-loading="false"
            data-state=""
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 13 22"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M6.938 5.5v4.813h4.812a.47.47 0 0 1 .438.437.45.45 0 0 1-.438.438H6.938V16a.45.45 0 0 1-.438.438.43.43 0 0 1-.437-.438v-4.812H1.25a.43.43 0 0 1-.437-.438.45.45 0 0 1 .437-.437h4.813V5.5a.45.45 0 0 1 .437-.437.47.47 0 0 1 .438.437"
                />
              </g>
              <defs>
                <clippath
                  id="49593f687d6be52155263aa4ff7f7bf0__a"
                >
                  <path
                    d="M0 0h13v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
